### PR TITLE
docs: added an example to use binary media types content handling

### PR DESCRIPTION
**Pull request by praneetap**
*Issue #, if available:*
Adding docs for a basic example on how to use binary media types and content handling
*Description of changes:*
docs for a basic example on how to use binary media types and content handling
*Description of how you validated changes:*
Deployed on cfn
*Checklist:*

- [x] `make pr` passes
- [x] Update documentation
- [x] Verify transformed template deploys and application functions as expected
- [x] Add/update example to `examples/2016-10-31`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: examples/2016-10-31/image_binary_media_types/template.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                  Fn::Sub: arn:${AWS::Partition}:apigateway:${AWS::Region}:lambda:path/2015-03-31/functions/${MyLambda.Arn}/invocations
+              responses: {}
+
+  MyLambda:
+    Type: AWS::Serverless::Function
+    Properties:
+      Handler: index.handler
+      Runtime: nodejs8.10
+      CodeUri:
+        Bucket: <YOUR_BUCKET>
```
Comment (keetonian) [inline examples/2016-10-31/image_binary_media_types/template.yaml]: Could you make this work with SAM CLI? Have this point to the local code package then add instructions to use `sam package` and `sam deploy` to deploy this application.

File: examples/2016-10-31/image_binary_media_types/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# API Gateway + Lambda Content gitsHandling Example
```
Comment (jlhood) [inline examples/2016-10-31/image_binary_media_types/README.md]: gitsHandling?

File: examples/2016-10-31/image_binary_media_types/README.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+Deploy the example into your account:
+
+```bash
+# Replace YOUR_S3_ARTIFACTS_BUCKET with the name of a bucket which already exists in your account
+aws cloudformation package --template-file template.yaml --output-template-file template.packaged.yaml --s3-bucket YOUR_S3_ARTIFACTS_BUCKET
+
+aws cloudformation deploy --template-file ./template.packaged.yaml --stack-name sam-example-api-lambda-image-content --capabilities CAPABILITY_IAM
+```
+
+Invoke the API's root endpoint `/none` to see the API respond with a 200 (assuming you followed the optional step above). You will find the response body contains a png file.
```
Comment (jlhood) [inline examples/2016-10-31/image_binary_media_types/README.md]: Why is the path `/none`? I think just `/` is more straightforward.

File: examples/2016-10-31/image_binary_media_types/README.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+```bash
+# Replace YOUR_S3_ARTIFACTS_BUCKET with the name of a bucket which already exists in your account
+aws cloudformation package --template-file template.yaml --output-template-file template.packaged.yaml --s3-bucket YOUR_S3_ARTIFACTS_BUCKET
+
+aws cloudformation deploy --template-file ./template.packaged.yaml --stack-name sam-example-api-lambda-image-content --capabilities CAPABILITY_IAM
+```
+
+Invoke the API's root endpoint `/none` to see the API respond with a 200 (assuming you followed the optional step above). You will find the response body contains a png file. 
+
+Make sure you set the client's request headers with "accept" : "image/png" to make this example work.
```
Comment (jlhood) [inline examples/2016-10-31/image_binary_media_types/README.md]: You should include a sample curl command to make it easier for customers to test this.

File: examples/2016-10-31/image_binary_media_types/template.yaml [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        basePath: "/none"
+        schemes:
+          - "https"
+        paths:
+          "/none":
+            get:
+              x-amazon-apigateway-integration:
+                httpMethod: POST
+                type: aws_proxy
+                contentHandling: CONVERT_TO_BINARY
```
Comment (jlhood) [inline examples/2016-10-31/image_binary_media_types/template.yaml]: SAM should allow users to specify this value without having to author their own swagger doc. One of the primary conveniences of using SAM is not having to author your own swagger. I don't think we can say SAM supports binary media types if you have to add a swagger doc to your template just to add this flag.

**Comment by keetonian** [comment]
Could update the template to the following:

```yaml
Transform: AWS::Serverless-2016-10-31

Globals:
  Api:
    # Send/receive binary data through the APIs
    BinaryMediaTypes:
      - image~1png
      - '*~1*' # forces the response to be in Binary. Otherwise, contentHandling: CONVERT_TO_BINARY is required in the swagger definition

Resources:
  MyLambda:
    Type: AWS::Serverless::Function
    Properties:
      Handler: index.handler
      Runtime: nodejs8.10
      CodeUri: ./src
      Events:
        None:
          Type: Api
          Properties:
            Method: get
            Path: /

Outputs:
  ApiUrl:
    Description: "API endpoint URL for Prod environment"
    Value:
      Fn::Sub: 'https://${ServerlessRestApi}.execute-api.${AWS::Region}.amazonaws.com/Prod/'
  MyLambda:
    Description: Lambda Function ARN
    Value:
      Fn::GetAtt: MyLambda.Arn
```

This is simpler, and can also use the AWS or SAM CLI to package and deploy the template:

```
sam package --s3-bucket <BUCKET_NAME> --output-template-file packaged.yaml

sam deploy --template-file packaged.yaml --stack-name <STACK_NAME> --capabilities CAPABILITY_IAM
```

**Comment by jlhood** [comment]
Next step: Will update this example once ContentHandling property is added (#553).

**Comment by jfuss** [comment]
Examples were removed/delete out of this repo. Closing

**Pull request closed without merging**